python/lsst/db/mysqlUtils.py

```diff
@@ -0,0 +1,387 @@
+    'read_default_file': 'defaults-file',
+    'charset':           'default-character-set',
+    'local_infile':      'local-infile'
+}
+# Map of mysql executable option names to MySQLdb driver connect() keywords.
+# Note: 'read_default_group' is not supported, since it can cause
+# the MySQLdb driver and the mysql executable to connect differently.
+optionToConnectArgMap = \
+    dict((v,k) for (k,v) in connectArgToOptionMap.iteritems())
+
+def readCredentialFile(fName, logger):
+    """
+    Reads all supported key/value pairs from fName and return a dictionary
+    containing these key/value pairs translated to names accepted by connect()
+    as needed). Hint, to get a subset, do something like:
+    dict = readCredentialFile(fN)
+    (hst, prt, usr, pwd) = [dict[k] for k in ('host', 'port', 'user', 'passwd')].
+    This function only reads from the [mysql] section, e.g., it is not full
+    equivalent to how mysql command like utility which obtains the value from the
+    last occurrence of k in section [mysql] or [client] in the file.
+    """
+    ret = {}
+    if fName.startswith('~'):
+        fName = os.path.expanduser(fName)
+    if not os.path.isfile(fName):
+        raise Exception("Required file '%s' not found" % fName)
+    cnf = ConfigParser.ConfigParser()
+    cnf.read(fName)
+
+    theSection = "mysql"
+    if not cnf.has_section(theSection):
+        raise Exception("Missing section '%s' in '%s'" % (theSection, fName))
+    for o in optionToConnectArgMap:
+        if cnf.has_option(theSection, o):
+            theKey = optionToConnectArgMap.get(o, o)
+            ret[theKey] = cnf.get(theSection, o)
+    logger.info("fetched %s from '%s' (password not shown)" % (
+            str(["%s:%s" % (x, ret[x]) for x in ret if not x == "passwd"]), fName))
+    return ret
+
+#### Database-related functions ####################################################
+def createDb(conn, dbName, mayExist=False):
+    """
+    Create database <dbName>.
+
+    @param dbName      Database name.
+    @param mayExist    Flag indicating what to do if the database exists.
+
+    Raises DbException if the database already exists and mayExist is False, or if
+    MySQL error occurs.
+
+    Note, it will not connect to that database and it will not make it default.
+    """
+    if dbName is None:
+        raise DbException(DbException.INVALID_DB_NAME, "<None>")
+    try:
+        conn.execute("CREATE DATABASE `%s`" % dbName)
+    except DBAPIError as e:
+        if e.orig[0] == MySqlErr.ER_DB_CREATE_EXISTS:
+            if not mayExist:
+                raise DbException(DbException.DB_EXISTS, dbName)
+        else:
+            raise DbException(DbException.MYSQL_ERROR, e.orig[0], e.orig[1])
+
+def useDb(conn, dbName):
+    """
+    Connect to database <dbName>.
+
+    @param dbName     Database name.
+
+    Raises DbException if MySQL error occurs.
+    """
+    if dbName is not None:
+        try:
+            conn.execute("USE `%s`" % dbName)
```

Comment:
I'm not sure what that means and how it could be done, but I want to withdraw my comment :) Input validation should be done on higher level API, not here.